lib/image_optim.rb

```diff
@@ -93,15 +104,21 @@ def workers_for_image(path)
   def optimize_image(original)
     original = ImagePath.convert(original)
     return unless (workers = workers_for_image(original))
-    result = Handler.for(original) do |handler|
+
+    cached = @cache.read(original)
+    return ImagePath::Optimized.new(cached, original) unless cached.nil?
```

Comment:
It may be shorter if you rename cached to `optimized` and run handler and write cache if nil is returned